code/datums/spells/lightning.dm

```diff
@@ -0,0 +1,89 @@
+/obj/effect/proc_holder/spell/targeted/lightning
+	name = "Lightning Bolt"
+	desc = "Throws a lightning bolt at the nearby enemy. Classic."
+	charge_type = "recharge"
+	charge_max	= 300
+	clothes_req = 1
+	invocation = "UN'LTD P'WAH!"
+	invocation_type = "shout"
+	range = 7
+	cooldown_min = 30
+	selection_type = "view"
+	random_target = 1
+	var/energy = 0
+	var/ready = 0
+
+/obj/effect/bolt
+	name = "Lightning bolt"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "lightning"
+	luminosity = 3
+
+/obj/effect/proc_holder/spell/targeted/lightning/Click()
+	if(!ready)
+		if(cast_check())
+			StartChargeup()
+	else
+		if(cast_check(skipcharge=1))
+			choose_targets()
+	return 1
+
+/obj/effect/proc_holder/spell/targeted/lightning/proc/StartChargeup(mob/user = usr)
+	ready = 1
+	user << "<span class='notice'>You start gathering the power.</span>"
+	//TODO: Add visual indictaor of charging
+	spawn(0)
+		while(ready)
+			sleep(1)
+			energy++
+			if(energy >= 100 && ready)
+				Discharge()
+
+/obj/effect/proc_holder/spell/targeted/lightning/proc/Discharge(mob/user = usr)
+	var/mob/living/M = user
+	M.electrocute_act(25,"Lightning Bolt")
+	M << "<span class='danger'>You lose control over the spell.</span>"
+	energy = 0
+	ready = 0
+	start_recharge()
+
+
+/obj/effect/proc_holder/spell/targeted/lightning/cast(list/targets, mob/user = usr)
+	if(!targets.len)
+		user << "<span class='notice'>No target found in range.</span>"
+		return
+
+	var/mob/living/carbon/target = targets[1]
```

Comment:
It's random, since ```random_target``` was also added (and set on this spell) in this PR.

Also yeah closest target should work.
```
var/mob/living/carbon/target
for(var/mob/living/L in targets)
    if(target && (get_dist(user,L) < get_dist(user,target)))
        target = L
    else
        target = L
```